internal/services/streamanalytics/stream_analytics_output_function_resource.go

```diff
@@ -0,0 +1,278 @@
+				}
+				return fmt.Errorf("reading %s: %+v", *id, err)
+			}
+
+			if props := resp.OutputProperties; props != nil && props.Datasource != nil {
+				v, ok := props.Datasource.AsAzureFunctionOutputDataSource()
+				if !ok {
+					return fmt.Errorf("converting output data source to a function output: %+v", err)
+				}
+
+				if v.FunctionAppName == nil || v.FunctionName == nil || v.MaxBatchCount == nil || v.MaxBatchSize == nil {
+					return nil
+				}
+
+				state := OutputFunctionResourceModel{
+					Name:               id.Name,
+					ResourceGroup:      id.ResourceGroup,
+					StreamAnalyticsJob: id.StreamingjobName,
+					FunctionApp:        *v.FunctionAppName,
+					FunctionName:       *v.FunctionName,
+					ApiKey:             metadata.ResourceData.Get("api_key").(string),
+					BatchMaxInBytes:    int(*v.MaxBatchSize),
+					BatchMaxCount:      int(*v.MaxBatchCount),
+				}
+				return metadata.Encode(&state)
+			}
+			return nil
+		},
+	}
+}
+
+func (r OutputFunctionResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.StreamAnalytics.OutputsClient
+			id, err := parse.OutputID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			var state OutputFunctionResourceModel
+			if err := metadata.Decode(&state); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			props := streamanalytics.Output{
+				Name: utils.String(state.Name),
+				OutputProperties: &streamanalytics.OutputProperties{
+					Datasource: &streamanalytics.AzureFunctionOutputDataSource{
+						Type: streamanalytics.TypeMicrosoftStorageTable,
+						AzureFunctionOutputDataSourceProperties: &streamanalytics.AzureFunctionOutputDataSourceProperties{
+							FunctionAppName: utils.String(state.FunctionApp),
+							FunctionName:    utils.String(state.FunctionName),
+							APIKey:          utils.String(state.ApiKey),
+							MaxBatchSize:    utils.Float(float64(state.BatchMaxInBytes)),
+							MaxBatchCount:   utils.Float(float64(state.BatchMaxCount)),
+						},
+					},
+				},
+			}
+
+			if _, err = client.Update(ctx, props, id.ResourceGroup, id.StreamingjobName, id.Name, ""); err != nil {
+				return fmt.Errorf("updating %s: %+v", *id, err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func (r OutputFunctionResource) Delete() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.StreamAnalytics.OutputsClient
+			id, err := parse.OutputID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			metadata.Logger.Infof("deleting %s", *id)
+
+			if resp, err := client.Delete(ctx, id.ResourceGroup, id.StreamingjobName, id.Name); err != nil {
+				if !response.WasNotFound(resp.Response) {
+					return fmt.Errorf("deleting %s: %+v", *id, err)
+				}
```

Comment:
this would indicate a bug, so can we remove this:

```suggestion
				return fmt.Errorf("deleting %s: %+v", *id, err)
```